platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/connectivity/TlsFactory.java

```diff
@@ -123,4 +107,40 @@ public TlsFactory(
         clientSocket.startHandshake();
         return clientSocket;
     }
+
+    @Override
+    public void handlePeerListUpdate(final @NonNull List<PeerInfo> peers) {
+        Objects.requireNonNull(peers);
+        try {
+            final KeyStore signingTrustStore = CryptoStatic.createPublicKeyStore(peers);
+
+            final char[] password = cryptoConfig.keystorePassword().toCharArray();
+            /* nondeterministic CSPRNG */
+            final SecureRandom nonDetRandom = CryptoStatic.getNonDetRandom();
+
+            // the agrKeyStore should contain an entry with both agrKeyPair.getPrivate() and agrCert
+            // PKCS12 uses file extension .p12 or .pfx
+            final KeyStore agrKeyStore = KeyStore.getInstance(CryptoConstants.KEYSTORE_TYPE);
+            agrKeyStore.load(null, null); // initialize
+            agrKeyStore.setKeyEntry("key", agrKey, password, new Certificate[] {agrCert});
+
+            // "PKIX" may be more interoperable than KeyManagerFactory.getDefaultAlgorithm or
+            // TrustManagerFactory.getDefaultAlgorithm(), which was "SunX509" on one system tested
+            final KeyManagerFactory keyManagerFactory =
+                    KeyManagerFactory.getInstance(CryptoConstants.KEY_MANAGER_FACTORY_TYPE);
+            keyManagerFactory.init(agrKeyStore, password);
+            final TrustManagerFactory trustManagerFactory =
+                    TrustManagerFactory.getInstance(CryptoConstants.TRUST_MANAGER_FACTORY_TYPE);
+            trustManagerFactory.init(signingTrustStore);
+            SSLContext.setDefault(sslContext);
```

Comment:
I think we should avoid setting the default, as we will have multiple instances of SSLContext

Reply:
it does make sense to set the default ssl context to that which belongs to the version we've chosen. The JVM may have different ssl context instances but we do* want the default to be for our chosen TLS version.

Anyway, how do you think we could end up with multiple instances?